test/libsolidity/semanticTests/viaYul/array_storage_index_access.sol

```diff
@@ -0,0 +1,87 @@
+contract C {
+	uint[] storageArray;
+	function test_indicies(uint256 len) public
+	{
+		storageArray.length = len;
+
+		for (uint i = 0; i < len; i++)
+			storageArray[i] = i + 1;
+
+		for (uint i = 0; i < len; i++)
+			require(storageArray[i] == i + 1);
+	}
+
+	function test_zeroed_indicies(uint256 len) public
+	{
+		storageArray.length = len;
+
+		for (uint i = 0; i < len; i++)
+			storageArray[i] = i + 1;
+
+		if (len > 3)
+		{
+			storageArray.length = 3;
+
+			for (uint i = 3; i < len; i++)
+			{
+				assembly {
+					mstore(0, storageArray_slot)
+					let pos := add(keccak256(0, 0x20), i)
+
+					if iszero(eq(sload(pos), 0)) {
+						revert(0, 0)
+					}
+				}
+			}
+
+		}
+
+		storageArray.length = 0;
+		storageArray.length = len;
+
+		for (uint i = 0; i < len; i++)
+		{
+			require(storageArray[i] == 0);
+
+			uint256 val = storageArray[i];
+			uint256 check;
+
+			assembly { check := iszero(val) }
+
+			require(check == 1);
+		}
+	}
+	function test_boundery_check(uint256 len, uint256 access) public returns
+(uint256)
+	{
+		storageArray.length = len;
+		return storageArray[access];
+	}
+}
+// ====
+// compileViaYul: true
+// ----
+// test_indicies(uint256): 1 ->
+// test_indicies(uint256): 129 ->
+// test_indicies(uint256): 5 ->
+// test_indicies(uint256): 10 ->
+// test_indicies(uint256): 15 ->
+// test_indicies(uint256): 0xFF ->
+// test_indicies(uint256): 1000 ->
+// test_indicies(uint256): 129 ->
+// test_indicies(uint256): 128 ->
+// test_indicies(uint256): 1 ->
+// test_zeroed_indicies(uint256): 1 ->
+// test_zeroed_indicies(uint256): 5 ->
+// test_zeroed_indicies(uint256): 10 ->
+// test_zeroed_indicies(uint256): 15 ->
+// test_zeroed_indicies(uint256): 0xFF ->
+// test_boundery_check(uint256, uint256): 10, 11 -> FAILURE
```

Comment:
Can you split this up into multiple files? On my screen, it is hard to see the function calls and the function definitions at the same time.